packages/core/core.ts

```diff
@@ -42,20 +41,23 @@ export function registerGlobals() {
 	globalThis.HookManager = hooking.hookManager;
 }
 
-export function initFuzzing(options: Options) {
+export async function initFuzzing(options: Options) {
 	registerGlobals();
 
-	options.customHooks.forEach((customHook) => {
-		importModule(customHook);
-	});
+	const importCustomHookPromises = options.customHooks.map((customHook) =>
+		importModule(customHook)
+	);
+	await Promise.all(importCustomHookPromises);
 
 	if (!options.dryRun) {
 		registerInstrumentor(options.includes, options.excludes);
 	}
 }
 
-function loadFuzzFunction(options: Options): fuzzer.FuzzFn {
-	const fuzzFn = importModule(options.fuzzTarget)[options.fuzzFunction];
+async function loadFuzzFunction(options: Options): Promise<fuzzer.FuzzFn> {
+	const fuzzFn = await importModule(options.fuzzTarget).then((res) => {
```

Comment:
I would tend to only `await` the call to `importModule` and then access the response synchronously and not in the `then` callback.

Reply:
Reverted this back to two const assignments. Could you clarify the benefit of doing so? Thought this was the cleaner solution.

Reply:
There is nothing wrong with the approach in itself. Chaining code using `then()` is just a way to wait for the resolution of a promise. To make this easier and get rid of long `then()` chains the `async/await` syntax was introduced. At this code point we used both `await` and `then` and that seems more complicated than it needs to be. If we can use `async/await` we probably should stickt to that.